app/teams/teams.go

```diff
@@ -196,3 +197,160 @@ func (ts *TeamService) RemoveTeamMember(teamMember *model.TeamMember) error {
 
 	return nil
 }
+
+func (ts *TeamService) RegenerateTeamInviteId(teamID string) (*model.Team, error) {
+	team, err := ts.GetTeam(teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	team.InviteId = model.NewId()
+
+	updatedTeam, err := ts.store.Update(team)
+	if err != nil {
+		return nil, err
+	}
+
+	ts.sendEvent(updatedTeam, model.WebsocketEventUpdateTeam)
+
+	return updatedTeam, nil
+}
+
+func (ts *TeamService) GetSchemeRolesForTeam(teamID string) (string, string, string, error) {
+	team, err := ts.GetTeam(teamID)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if team.SchemeId != nil && *team.SchemeId != "" {
+		scheme, err := ts.schemeStore.Get(*team.SchemeId)
+		if err != nil {
+			return "", "", "", err
+		}
+		return scheme.DefaultTeamGuestRole, scheme.DefaultTeamUserRole, scheme.DefaultTeamAdminRole, nil
+	}
+
+	return model.TeamGuestRoleId, model.TeamUserRoleId, model.TeamAdminRoleId, nil
+}
+
+func (ts *TeamService) UpdateTeamMemberRoles(teamID string, userID string, newRoles string) (*model.TeamMember, error) {
+	member, err := ts.store.GetMember(context.Background(), teamID, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	if member == nil {
+		return nil, NotTeamMemberError
+
+	}
+
+	schemeGuestRole, schemeUserRole, schemeAdminRole, err := ts.GetSchemeRolesForTeam(teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	prevSchemeGuestValue := member.SchemeGuest
+
+	var newExplicitRoles []string
+	member.SchemeGuest = false
+	member.SchemeUser = false
+	member.SchemeAdmin = false
+
+	for _, roleName := range strings.Fields(newRoles) {
+		var role *model.Role
+		role, err = ts.roleStore.GetByName(context.Background(), roleName)
+		if err != nil {
+			// err.StatusCode = http.StatusBadRequest
```

Comment:
Was this a debug line?

Reply:
Uh, it escaped from me. It's not a debug line btw it was updating the appErr's status code for the app-side implementation.